x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/dataframe/DataFrameAnalyticsConfig.java

```diff
@@ -170,34 +147,58 @@ public String getDest() {
         return analyses;
     }
 
-    @Nullable
-    public Map<String, Object> getQuery() {
-        return query;
-    }
-
-    @Nullable
-    public QueryBuilder getParsedQuery() {
-        return querySupplier.get();
+    /**
+     * Get the fully parsed query from the semi-parsed stored {@code Map<String, Object>}
+     *
+     * @param namedXContentRegistry XContent registry to transform the lazily parsed query
+     * @return Fully parsed query
+     */
+    public QueryBuilder getParsedQuery(NamedXContentRegistry namedXContentRegistry) {
```

Comment:
This is not necessary `getParsedQuery` should be something like:
```
public QueryBuilder getParsedQuery() {
    if(queryProvider == null) { // I am unsure around null guarantees  
        return null;
    }
    if(queryProvider.getParsingException() != null) {
        throw new ElasticSearchException(queryProvider.getParsingException());
    }
    return queryProvider.getParsedQuery();
}
```
Since we are not serializing against a previous version of this object that did not have the `QueryProvider` object.